components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/CertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtErrors;
+import org.wso2.carbon.identity.certificate.management.core.dao.impl.CertificateManagementDAOImpl;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtClientException;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.core.model.Certificate;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateMgtExceptionHandler;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateValidator;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.util.UUID;
+
+/**
+ * This class is responsible for managing trusted certificates of a tenant.
+ */
+public class CertificateManagementServiceImpl implements CertificateManagementService {
+
+    private static final Log LOG = LogFactory.getLog(CertificateManagementServiceImpl.class);
+    private static final CertificateManagementService INSTANCE = new CertificateManagementServiceImpl();
+    private static final CertificateManagementDAOImpl DAO = new CertificateManagementDAOImpl();
+    private static final CertificateValidator CERTIFICATE_VALIDATOR = new CertificateValidator();
```

Comment:
Make the methods of `CertificateValidator` class static instead of creating a static object since this is a Utils class. Change other occurrences as well.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6043/commits/865201e1a31faf80eb1768629ad74833eafe851b